crates/contract/src/update.rs

```diff
@@ -175,6 +179,17 @@ impl ProposedUpdates {
         }
         Some(promise)
     }
+
+    /// Removes the vote for [`AccountId`]
+    pub fn remove_vote(&mut self, voter: &AccountId) {
+        if let Some(previous_id) = self.vote_by_participant.remove(voter) {
+            if let Some(entry) = self.entries.get_mut(&previous_id) {
+                entry.votes.remove(voter);
```

Comment:
Shouldn't we remove `entry` from `self.entries` if no votes remain after this removal? 🤔